pandas/core/generic.py

```diff
@@ -9656,11 +9638,30 @@ def describe_categorical_1d(data):
 
             return pd.Series(result, index=names, name=data.name, dtype=dtype)
 
+        def describe_timestamp_1d(data):
+            # GH-30164
+            tz = data.dt.tz
+            asint = data.dropna().values.view("i8")
```

Comment:
you don't need the .values.view, just use` len(data.dropna())` for the shape check

data.min() and so on all work and return NaT correctly, none of these gymnastics are needed

Reply:
I know that's what the current code does, but let's simplify